### PR TITLE
gitreceive,test: Include current env in env dir

### LINKED ISSUE #3426: Failed Launch when deploying Rails App (Digital Ocean and AWS)
I have tried deploying a Rails app with Flynn and it failed on both AWS and Digital Ocean. 
Here is the error that occurred in both cases.
```
Running: rake assets:precompile
       rake aborted!
       ArgumentError: Missing required arguments: aws_access_key_id, aws_secret_access_key
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/fog-core-1.42.0/lib/fog/core/service.rb:244:in `validate_options'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/fog-core-1.42.0/lib/fog/core/service.rb:268:in `handle_settings'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/fog-core-1.42.0/lib/fog/core/service.rb:98:in `new'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/fog-core-1.42.0/lib/fog/core/services_mixin.rb:16:in `new'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/fog-core-1.42.0/lib/fog/storage.rb:27:in `new'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/carrierwave-0.11.2/lib/carrierwave/uploader/configuration.rb:83:in `eager_load_fog'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/carrierwave-0.11.2/lib/carrierwave/uploader/configuration.rb:96:in `fog_credentials='
       /tmp/build/config/initializers/carrier_wave.rb:3:in `block in <top (required)>'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/carrierwave-0.11.2/lib/carrierwave/uploader/configuration.rb:118:in `configure'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/carrierwave-0.11.2/lib/carrierwave.rb:14:in `configure'
       /tmp/build/config/initializers/carrier_wave.rb:2:in `<top (required)>'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/activesupport-5.0.0.1/lib/active_support/dependencies.rb:287:in `load'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/activesupport-5.0.0.1/lib/active_support/dependencies.rb:287:in `block in load'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/activesupport-5.0.0.1/lib/active_support/dependencies.rb:259:in `load_dependency'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/activesupport-5.0.0.1/lib/active_support/dependencies.rb:287:in `load'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/railties-5.0.0.1/lib/rails/engine.rb:648:in `block in load_config_initializer'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/activesupport-5.0.0.1/lib/active_support/notifications.rb:166:in `instrument'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/railties-5.0.0.1/lib/rails/engine.rb:647:in `load_config_initializer'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/railties-5.0.0.1/lib/rails/engine.rb:612:in `block (2 levels) in <class:Engine>'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/railties-5.0.0.1/lib/rails/engine.rb:611:in `each'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/railties-5.0.0.1/lib/rails/engine.rb:611:in `block in <class:Engine>'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/railties-5.0.0.1/lib/rails/initializable.rb:30:in `instance_exec'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/railties-5.0.0.1/lib/rails/initializable.rb:30:in `run'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/railties-5.0.0.1/lib/rails/initializable.rb:55:in `block in run_initializers'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/railties-5.0.0.1/lib/rails/initializable.rb:44:in `each'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/railties-5.0.0.1/lib/rails/initializable.rb:44:in `tsort_each_child'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/railties-5.0.0.1/lib/rails/initializable.rb:54:in `run_initializers'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/railties-5.0.0.1/lib/rails/application.rb:352:in `initialize!'
       /tmp/build/config/environment.rb:5:in `<top (required)>'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/railties-5.0.0.1/lib/rails/application.rb:328:in `require'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/railties-5.0.0.1/lib/rails/application.rb:328:in `require_environment!'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/railties-5.0.0.1/lib/rails/application.rb:448:in `block in run_tasks_blocks'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/sprockets-rails-3.1.1/lib/sprockets/rails/task.rb:62:in `block (2 levels) in define'
       /tmp/build/vendor/bundle/ruby/2.3.0/gems/rake-11.2.2/exe/rake:27:in `<top (required)>'
       Tasks: TOP => environment
       (See full trace by running task with --trace)
       !
       !     Precompiling assets failed.
       !
ERROR: Build failed: exec: job exited with status 1
exit status 1
```

I set the credentials which it shows are missing inside a file: _project_root/config/initializers/carrier_wave.rb_

```
if Rails.env.production?
  CarrierWave.configure do |config|
    config.fog_credentials = {
      # Configuration for Amazon S3
      provider: 'AWS',
      aws_access_key_id: ENV['S3_ACCESS_KEY'],
      aws_secret_access_key: ENV['S3_SECRET_KEY'],
      region: ENV['S3_REGION']
    }
    config.fog_directory = ENV['S3_BUCKET']
  end
end
```

And just before clicking launch on the Flynn dashboard, I enter all the required Environment Variables. I have this project working perfectly on heroku. Any idea if I am doing anything wrong?

### PR DESCRIPTION
**Pull request by jvatic**
Closes #3426

### CODE REVIEW

File: test/apps/ruby-compile-env/Rakefile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+task 'assets:precompile' do
+  if ENV['FOO'] != 'BAR'
```
Comment (lmars) [inline test/apps/ruby-compile-env/Rakefile]: You could just use the `env-dir` test app which prints the value of `FOO` on compile (see [here](https://github.com/flynn/flynn/blob/master/test/apps/env-dir/bin/compile#L10)), and then test `flynn-receiver -e FOO=BAR` prints `BAR` (you will need to use the inline buildpack like [this test](https://github.com/flynn/flynn/blob/master/test/test_git_deploy.go#L38)).

File: test/test_gitreceive.go [hunk 2]
```diff
@@ -53,3 +60,63 @@ func (s *GitreceiveSuite) TestSlugbuilderLimit(t *c.C) {
...
+	t.Assert(err, c.IsNil)
+	defer data.Close()
+
+	env := map[string]string{
+		"FOO": "BAR",
+	}
+	meta := map[string]string{
+		"git": "true",
+	}
+	s.deployTar(t, app, "test-rev", env, meta, data)
```
Comment (lmars) [inline test/test_gitreceive.go]: You could simplify this test by using `tar` and the CLI from a shell:

```
tar c -C test/apps/env-dir . | flynn -a gitreceive run /bin/flynn-receiver APP REV -e FOO=BAR
```

If you do go with that, you'll probably want to use `bash -o pipefail -e -c "command"` to make sure it exits non-zero on failure.

**Comment by jvatic** [comment]
@lmars comments addressed.

**Comment by jvatic** [comment]
Fixed test

**Pull request merged**